packages/veritone-widgets/src/widgets/EnginePicker/MultiplePicker.js

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { guid } from '../../shared/util';
+
+import { any, bool, func, objectOf } from 'prop-types';
+
+import * as multipleEngineSelectionModule from '../../redux/modules/multipleEngineSelection';
+
+import widget from '../../shared/widget';
+import EnginePicker from '.';
+import SelectionInfoPanel from './SelectionInfoPanel';
+
+@connect(
+  (state, { ids }) => {
+    return {
+      selected: multipleEngineSelectionModule.selectedEngines(state, ids),
+      canSelectMore: multipleEngineSelectionModule.canSelectMore(state)
+    };
+  },
+  {
+    toggleSelection: multipleEngineSelectionModule.toggleEngine
+  }
+)
+class MultipleEnginePicker extends React.Component {
+  static propTypes = {
+    canSelectMore: bool,
+    selected: objectOf(any),
+    toggleSelection: func
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.id = guid();
```

Comment:
i'm thinking more descriptive property names would help here? just to make the code more self-documenting